File: patch_multiselect.py
import sys

with open('src/features/timing/TimingPage.jsx', 'r') as f:
    content = f.read()

# Define the exact block to move
start_marker = '  const ZOMATO_BLUE = "#2368ee";'
end_marker = '  // --- MultiSelect Component ---'

# Find the start of the block
start_idx = content.find(start_marker)
if start_idx == -1:
    print("Could not find start marker")
    sys.exit(1)

# Find the end of the block, which is the closing brace of MultiSelect
# I'll find it by searching for the exact line '  };\n\n  return ('
end_idx = content.find('  };\n\n  return (', start_idx)
if end_idx == -1:
    print("Could not find end marker")
    sys.exit(1)

# Include '  };\n\n' in the extracted block to remove it cleanly
block_to_move = content[start_idx : end_idx + 6] 

# Remove the block from the original location
new_content = content[:start_idx] + content[end_idx + 6:]

# Now, we insert the block just before 'export default function TimingPage() {'
target_str = 'export default function TimingPage() {'
target_idx = new_content.find(target_str)
if target_idx == -1:
    print("Could not find target export")
    sys.exit(1)

# Fix indentation: remove the leading two spaces from every line in the block
lines = block_to_move.split('\n')
fixed_lines = [line[2:] if line.startswith('  ') else line for line in lines]
fixed_block = '\n'.join(fixed_lines)

final_content = new_content[:target_idx] + fixed_block + target_str + new_content[target_idx + len(target_str):]

with open('src/features/timing/TimingPage.jsx', 'w') as f:
    f.write(final_content)

print("Successfully moved MultiSelect out of TimingPage!")

File: patch_multiselect.js
import fs from 'fs';

let content = fs.readFileSync('src/features/timing/TimingPage.jsx', 'utf8');

// 1. Locate the exact block
const startIndex = content.indexOf('  const ZOMATO_BLUE = "#2368ee";');
const endMarker = '  };';
// find the first '  };' after startIndex + 100
let endIndex = content.indexOf(endMarker, startIndex + 100);

if (startIndex !== -1 && endIndex !== -1) {
  // we want to include the '  };\n'
  const blockToMove = content.substring(startIndex, endIndex + endMarker.length + 1);
  
  // 2. Remove the block from the original location
  content = content.substring(0, startIndex) + content.substring(endIndex + endMarker.length + 1);
  
  // 3. Insert the block just before `export default function TimingPage() {`
  const targetIndex = content.indexOf('export default function TimingPage() {');
  
  // clean up the indentation of the block for the root level
  const cleanedBlock = blockToMove.split('\n').map(line => line.replace(/^  /, '')).join('\n');
  
  content = content.substring(0, targetIndex) + cleanedBlock + '\n' + content.substring(targetIndex);
  
  fs.writeFileSync('src/features/timing/TimingPage.jsx', content);
  console.log("Successfully moved MultiSelect out of TimingPage!");
} else {
  console.error("Could not find the block to move.");
}
